### PR TITLE
Fixed peer generation

### PR DESCRIPTION
**Pull request by evabishchevich**
This change is [Reviewable](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/1651)

### CODE REVIEW

File: src/android/java/io/jxcore/node/JXcoreExtension.java [hunk 1]
```diff
@@ -512,7 +512,11 @@ public static void notifyPeerAvailabilityChanged(PeerProperties peerProperties,
 
         try {
             jsonObject.put(EVENT_VALUE_PEER_ID, peerProperties.getId());
-            jsonObject.put(EVENT_VALUE_PEER_GENERATION, peerProperties.getExtraInformation());
+            Integer gen = peerProperties.getExtraInformation();
+            if (!isAvailable) {
+                gen = (peerProperties.getExtraInformation() == 0) ? null : peerProperties.getExtraInformation();
```
Comment (larryonoff) [inline src/android/java/io/jxcore/node/JXcoreExtension.java]: is it valid calling one method twice ? what about just cache value and check cached value?

**Comment by evabishchevich** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/android/java/io/jxcore/node/JXcoreExtension.java, line 517 at r1](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1651#-KZ6Cyb1Pl4Z-DbYfBG0-r1-517:-KZ7INxQt7uVAtVFbnH_:b8cmfve) ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/68fa1489e8cf8fbfabb5da91c3bcacd51244fe02/src/android/java/io/jxcore/node/JXcoreExtension.java#L517)):*
<details><summary><i>Previously, larryonoff (larryonoff) wrote…</i></summary><blockquote>

is it valid calling one method twice ? what about just cache value and check cached value?
</blockquote></details>

It's okay. It is just getting int via getter. We don't need one more variable. If talk about performance, current solutions consumes less memory

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1651)*

**Comment by larryonoff** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1651)*

**Comment by larryonoff** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/1651#-:-KZ7JD_RmA3VUZY685OA:bnfp4nl)*

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (e1b00a5)

**Comment by ThaliTester** [comment]
Test 98312760 (e1b00a5) build started.

**Comment by ThaliTester** [comment]
Test 98312760 (e1b00a5) build started.

**Comment by ThaliTester** [comment]
Test (Success) 98312760 build is completed (e1b00a5)

See https://github.com/ThaliTester/TestResults/tree/98312760e1b00a5_Fixed_peer_generation_evabishchevich/ for the logs

**Comment by ThaliTester** [comment]
Test 98312760e1b00a5(e1b00a5) has failed

See https://github.com/ThaliTester/TestResults/tree/98312760e1b00a5_Fixed_peer_generation_evabishchevich/ for the fail logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (b2c4df9)

**Comment by ThaliTester** [comment]
Test 98312760 (b2c4df9) build started.

**Comment by ThaliTester** [comment]
Test (Success) 98312760 build is completed (b2c4df9)

See https://github.com/ThaliTester/TestResults/tree/98312760b2c4df9_Fixed_peer_generation_evabishchevich/ for the logs

**Comment by ThaliTester** [comment]
Test 98312760b2c4df9(b2c4df9) has failed

See https://github.com/ThaliTester/TestResults/tree/98312760b2c4df9_Fixed_peer_generation_evabishchevich/ for the fail logs

**Comment by evabishchevich** [comment]
@yaronyg There are CI issues, not the code problems. I'm going to merge this PR

**Pull request merged**